wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+                                                          metadata=admin_role_binding_name)
+        auth_api.create_namespaced_role_binding(namespace=namespace, body=admin_role_binding)
+
+        # adding image-puller role binding
+        puller_role = self.kclient.V1ObjectReference(name='system:image-puller')
+        group_name = 'system:serviceaccounts:{proj}'.format(proj=namespace)
+        puller_group = self.kclient.V1ObjectReference(name=group_name,
+                                                      kind='SystemGroup',
+                                                      namespace=namespace)
+        role_binding_name = self.kclient.V1ObjectMeta(name='image-puller-binding')
+        puller_role_binding = self.ociclient.V1RoleBinding(role_ref=puller_role,
+                                                           subjects=[puller_group],
+                                                           metadata=role_binding_name)
+        auth_api.create_namespaced_role_binding(namespace=namespace, body=puller_role_binding)
+
+    def delete_project(self, name, wait=120):
+        """Removes project(namespace) and all entities in it.
+
+        Args:
+            name: project name
+            wait: within this time project should disappear
+        Return: None
+        """
+        self.logger.info("removing project %s", name)
+        if self.does_project_exist(name=name):
+            self.o_api.delete_project(name=name)
+            if not self.does_project_exist(name=name, wait=wait):
+                raise TimedOutError('project {n} was not removed within {w} sec'.format(n=name,
+                                                                                        w=wait))
+
+    def scale_entity(self, namespace, name, replicas, wait=60):
+        """Allows to scale up/down entities.
+        One of cases when this is necessary is emulation of stopping/starting appliance
+
+        Args:
+            namespace: openshift namespace
+            name: entity name. it can be either stateless Pod from DeploymentConfig or StatefulSet
+            replicas: number of replicas 0..N
+            wait: time to wait for scale up/down
+        Return: None
+        """
+        # only dc and statefulsets can be scaled
+        st_api = self.kclient.AppsV1beta1Api()
+
+        scale_val = self.kclient.V1Scale(spec=self.kclient.V1ScaleSpec(replicas=replicas))
+        if self.is_deployment_config(name=name, namespace=namespace):
+            self.o_api.patch_namespaced_deployment_config_scale(name=name, namespace=namespace,
+                                                                body=scale_val)
+
+            def check_scale_value():
+                got_scale = self.o_api.read_namespaced_deployment_config_scale(name=name,
+                                                                               namespace=namespace)
+                return 0 if got_scale.spec.replicas is None else int(got_scale.spec.replicas)
```

Comment:
**OPTIONAL**: Looking at this ternary, assuming `.replicas` is `str` or `None`, you could shorten it like this:

```python
return int(got_scale.spec.replicas or 0)
```

Just an idea :)